### PR TITLE
fix(log): fix error logging for long queries

### PR DESCRIPTION
**Pull request by bziobrowski**
Changes : 
- Improve error logging for queries exceeding LogRecordSink buffer size by : 
  - leaving space for EOL chars (to prevent long log message being truncated and joined with following line)
  - logging query text after error message & errno 
- improve logging of stack overflow error thrown from regex_replace matching a very long string

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 37 (75.68%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|9|17|52.94%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 40 / 49 (81.63%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateCrashFunctionFactory.java|6|12|50.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|15|17|88.24%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java [hunk 1]
```diff
@@ -172,6 +175,14 @@ public boolean isRuntimeConstant() {
         public void toPlan(PlanSink sink) {
             sink.val("regexp_replace(").val(value).val(',').val(pattern).val(',').val(replacement).val(')');
         }
+
+        private boolean find() {
+            try {
+                return matcher.find();
+            } catch (StackOverflowError err) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java:182]: this looks a bit crazy, is this a normal patten of behaviour of regex?

Comment (bziobrowski) [reply core/src/main/java/io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java:182]: It is possible for certain patterns and very long input strings  due to excessive backtracking .

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 43 / 50 (86.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateCrashFunctionFactory.java|6|12|50.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 43 / 50 (86.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/str/CharSink.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/SimulateCrashFunctionFactory.java|6|12|50.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|17|17|100.00%|
|:large_blue_circle:|io/questdb/log/LogRecordSink.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|

**Pull request merged**